backend/src/common/services/azure/azure-ai.service.ts

```diff
@@ -0,0 +1,176 @@
+        this.azureSubscriptionKey = config.ENV.AZURE.AZURE_SERVICE_KEY;
+        this.azureEndpoint = config.ENV.AZURE.AZURE_SERVICE_ENDPOINT;
+    }
+
+    public async getAvatarVoices(): Promise<GetAvatarVoicesResponseDto[]> {
+        const url = `https://${this.config.ENV.AZURE.AZURE_SERVICE_REGION}.tts.speech.microsoft.com/cognitiveservices/voices/list`;
+        const options = {
+            method: HTTPMethod.GET,
+            headers: {
+                'Ocp-Apim-Subscription-Key':
+                    this.config.ENV.AZURE.AZURE_SERVICE_KEY,
+            },
+        };
+        const response = await fetch(url, options);
+        if (!response.ok) {
+            throw new HttpError({
+                message: response.statusText,
+                status: HttpCode.INTERNAL_SERVER_ERROR,
+            });
+        }
+        const data = await response.json();
+        return data
+            .filter(
+                (data: GetAvatarVoicesResponseDto) => data.Locale === 'en-US',
+            )
+            .map((data: GetAvatarVoicesResponseDto) => ({
+                DisplayName: data.DisplayName,
+                ShortName: data.ShortName,
+                Locale: data.Locale,
+                Gender: data.Gender,
+                LocaleName: data.LocaleName,
+                StyleList: data.StyleList,
+            }));
+    }
+
+    public async textToSpeech(
+        text: string,
+        voiceName: string,
+    ): Promise<string> {
+        const audioFileName = `ttsConverted_${Date.now()}.wav`;
+        const speechConfig = SpeechSDK.SpeechConfig.fromSubscription(
+            this.azureSubscriptionKey,
+            this.azureRegion,
+        );
+        const audioConfig =
+            SpeechSDK.AudioConfig.fromAudioFileOutput(audioFileName);
+        speechConfig.speechSynthesisLanguage = voiceName
+            .split('-', 2)
+            .join('-');
+        speechConfig.speechSynthesisVoiceName = voiceName;
+        const speechSynthesizer = new SpeechSDK.SpeechSynthesizer(
+            speechConfig,
+            audioConfig,
+        );
+        await new Promise<void>((resolve, reject) => {
+            speechSynthesizer.speakTextAsync(
+                text,
+                (result) => {
+                    if (
+                        result.reason ===
+                        SpeechSDK.ResultReason.SynthesizingAudioCompleted
+                    ) {
+                        resolve();
+                    } else {
+                        reject(
+                            new Error(
+                                `Speech synthesis canceled: ${result.errorDetails}`,
+                            ),
+                        );
+                    }
+                    speechSynthesizer.close();
+                },
+                (error) => {
+                    speechSynthesizer.close();
+                    reject(error);
+                },
+            );
+        });
```

Comment:
Don't mix async-await style with promises

Reply:
Promise moved to a privet method to be called and in public textToSpeech method only async-await